### PR TITLE
[DOM-277] Get Domain push feature status endpoint implementation

### PR DESCRIPTION
**Pull request by ms-darianbenito**
This PR includes the following implementation:
image

Pendings:
- [x] Some extra tests

Related to https://makingsense.atlassian.net/browse/DOM-277

### CODE REVIEW

**Comment by andresmoschini** [comment]
👍

File: Doppler.PushContact.Test/Controllers/DomainControllerTest.cs [hunk 1]
```diff
@@ -193,5 +193,132 @@ public async Task Upsert_should_return_internal_server_error_when_service_throw_
...
+
+            // Act
+            var response = await client.SendAsync(request);
+            _output.WriteLine(response.GetHeadersAsString());
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var isPushFeatureEnabledResponse = await response.Content.ReadAsStringAsync();
+            Assert.Equal(expectedDomain.IsPushFeatureEnabled.ToString(), isPushFeatureEnabledResponse);
+        }
```
Comment (andresmoschini) [inline Doppler.PushContact.Test/Controllers/DomainControllerTest.cs:290]: Hablamos esto con Darian

```suggestion
        [Theory]
        [InlineData(true, "True")]
        [InlineData(false, "False")]
        public async Task GetPushFeatureStatus_should_response_push_feature_status_returned_by_domain_service(bool isPushFeatureEnabledValue, string expectedResponse)
        {
            // Arrange
            var fixture = new Fixture();

            var name = fixture.Create<string>();
            var domain = fixture.Create<Domain>();
            var domain.IsPushFeatureEnabled = isPushFeatureEnabledValue;

            var domainServiceMock = new Mock<IDomainService>();
            domainServiceMock.Setup(x => x.GetByNameAsync(name))
                .ReturnsAsync(domain);

            var client = _factory.WithWebHostBuilder(builder =>
            {
                builder.ConfigureTestServices(services =>
                {
                    services.AddSingleton(domainServiceMock.Object);
                });

            }).CreateClient(new WebApplicationFactoryClientOptions());

            var request = new HttpRequestMessage(HttpMethod.Get, $"domains/{name}/isPushFeatureEnabled");

            // Act
            var response = await client.SendAsync(request);
            _output.WriteLine(response.GetHeadersAsString());

            // Assert
            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
            var isPushFeatureEnabledResponse = await response.Content.ReadAsStringAsync();
            Assert.Equal(expectedResponse, isPushFeatureEnabledResponse);
        }
```

File: Doppler.PushContact.Test/Controllers/DomainControllerTest.cs [hunk 2]
```diff
@@ -193,5 +193,135 @@ public async Task Upsert_should_return_internal_server_error_when_service_throw_
...
+            var request = new HttpRequestMessage(HttpMethod.Get, $"domains/{name}/isPushFeatureEnabled");
+
+            // Act
+            var response = await client.SendAsync(request);
+            _output.WriteLine(response.GetHeadersAsString());
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var isPushFeatureEnabledResponse = await response.Content.ReadAsStringAsync();
+            Assert.Equal(expectedIsPushFeatureEnabled, isPushFeatureEnabledResponse);
```
Comment (andresmoschini) [inline Doppler.PushContact.Test/Controllers/DomainControllerTest.cs:292]: In my opinion, `expectedResponse` would be more semantically correct.

```suggestion
            Assert.Equal(expectedResponse, isPushFeatureEnabledResponse);
```

File: Doppler.PushContact.Test/Controllers/DomainControllerTest.cs (hunk 1 above)
Comment (ms-darianbenito) [reply Doppler.PushContact.Test/Controllers/DomainControllerTest.cs]: Buena idea! Gracias!
_Hay una diferencia en los valores de expectedResponse, ambos valores posibles son en minúsculas_

Comment (andresmoschini) [reply Doppler.PushContact.Test/Controllers/DomainControllerTest.cs]: Joya, quiere decir que el test sirve!

**Pull request merged**